### PR TITLE
Fix base_score extraction from JSON

### PR DESCRIPTION
**Pull request by imbsxav**
Updated base_score extraction to handle string format.

I tested the curent python code on a simple example (a dummy xgboost mode) and it would crash the fast forest C++ text parsing as it contains base_score and brackets ([]) on the same line (which the C++ parsing code doesn't support).

### CODE REVIEW

**Comment by guitargeek** [comment]
Thank you very much! I had to change a bit more, to cover the general case where they are different base scores for the different forests in a softmax ensemble: 557e523f71bc632599877af37bcaccc8f66cfb27.

But as you can see in the CI, everything works now again with the newest XGBoost release! I'll also tag a new fastforest release now.

**Comment by guitargeek** [comment]
https://github.com/guitargeek/XGBoost-FastForest/releases/tag/v0-6-0
